src/main/java/fi/vrk/xroad/monitor/actor/Supervisor.java

```diff
@@ -0,0 +1,122 @@
+package fi.vrk.xroad.monitor.actor;
+
+import akka.actor.AbstractActor;
+import akka.actor.ActorRef;
+import akka.actor.OneForOneStrategy;
+import akka.actor.SupervisorStrategy;
+import akka.japi.pf.DeciderBuilder;
+import akka.pattern.Patterns;
+import akka.routing.SmallestMailboxPool;
+import akka.util.Timeout;
+import fi.vrk.xroad.monitor.extensions.SpringExtension;
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import scala.concurrent.Await;
+import scala.concurrent.duration.Duration;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static akka.actor.SupervisorStrategy.resume;
+import static fi.vrk.xroad.monitor.util.MonitorCollectorConstants.SUPERVISOR_MONITOR_DATA_ACTOR_POOL_SIZE;
+
+/**
+ * Supervisor for actors
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class Supervisor extends AbstractActor {
+
+  @Getter
+  private ActorRef resultCollectorActor;
+  private ActorRef monitorDataRequestPoolRouter;
+
+  @Autowired
+  SpringExtension ext;
+
+  public Supervisor (ActorRef resultCollectorActor, ActorRef monitorDataRequestPoolRouter) {
+    this.resultCollectorActor = resultCollectorActor;
+    this.monitorDataRequestPoolRouter = monitorDataRequestPoolRouter;
+  }
+
+  public Supervisor () {}
+
+  @Override
+  public void preStart() throws Exception {
+    log.info("preStart");
+
+    if (resultCollectorActor == null) {
+      resultCollectorActor = getContext().actorOf(ext.props("resultCollectorActor"));
+    }
+
+    if (monitorDataRequestPoolRouter == null) {
+      monitorDataRequestPoolRouter = getContext()
+              .actorOf(new SmallestMailboxPool(SUPERVISOR_MONITOR_DATA_ACTOR_POOL_SIZE)
+                      .props(ext.props("monitorDataActor", resultCollectorActor)));
+    }
+
+    super.preStart();
+  }
+
+  @Override
+  public Receive createReceive() {
+    log.info("createReceive");
+    return receiveBuilder()
+        .match(StartCollectingMonitorDataCommand.class, this::handleMonitorDataRequest)
+        .matchAny(obj -> log.error("Unhandled message: {}", obj))
+        .build();
+  }
+
+  private void handleMonitorDataRequest (StartCollectingMonitorDataCommand request) {
+    Timeout timeout = new Timeout(1, TimeUnit.MINUTES);
+
+    try {
+      Await.ready(Patterns.ask(resultCollectorActor, request.getSecurityServerInfos(), timeout), timeout.duration());
+    } catch (TimeoutException | InterruptedException e) {
+      log.error("Failed to initialize the ResultCollectorActor, {}", e);
```

Comment:
Should not continue (and delegate to monitorDataActor pool) if resultCollector failed to initialize. Should just fail and stop, I think.